changelog.d/866.bugfix

```diff
@@ -0,0 +1 @@
+Topic changes from Matrix no longer cause a ghost user to join the room.
```

Comment:
I'm failing to see how this relates to the rest of the described problem?

Reply:
The changelog describes the fix from a user PoV. The issue mentioned is poorly explained. I'll update the description of this PR for historical reasons.

Reply:
Makes sense now, thanks!